### PR TITLE
Test: Add comprehensive coverage for AbandonedProblemsService #8

### PR DESCRIPTION
**Pull request by Mohammed2372**
**Description: Added unit tests Including:**
- Core Logic Tests: Verified identification of abandoned vs. solved-later problems (`test_analysis.py`).
- Statistics Tests: Covered aggregation by tags and rating bins (`test_stats.py`).
- Edge Cases: Handled empty submissions, unrated problems, and disjoint tag sets.
- Test Configuration: Added `__init__.py` to the test package to ensure correct path resolution for the backend module during test collection.

### CODE REVIEW

File: backend/tests/unit/abandoned_problems_service/__init__.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import sys
+import os
+
+# Get the path to the project root (betterforces)
+current_dir = os.path.dirname(os.path.abspath(__file__))
+# Go up 4 levels: abandoned_problems_service -> unit -> tests -> backend -> betterforces
+project_root = os.path.abspath(os.path.join(current_dir, "../../../.."))
```
Comment (deyna256) [inline backend/tests/unit/abandoned_problems_service/__init__.py]: Why do we need to go up to `betterforces` level? Do we really need this file?

File: backend/tests/unit/abandoned_problems_service/test_stats.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        ([["graphs"], ["strings"]], {"graphs": 1, "strings": 1}),
+        # Case 3: Same tag repeated across 3 problems
+        ([["math"], ["math"], ["math"]], {"math": 3}),
+    ],
+)
+def test_aggregation_by_tags(mock_submission, problems_tags, expected_counts) -> None:
+    """Test if abandoned problem are correctly grouped by tags."""
+
+    submissions = []
+
```
Comment (deyna256) [inline backend/tests/unit/abandoned_problems_service/test_stats.py]: too many blank lines, we want to make our tests bodies more compact

**Review by deyna256 — APPROVED** [review]
@Mohammed2372 cool! thank you!
